### PR TITLE
Refactoring to complete new solution

### PR DESCRIPTION
**Pull request by Steph186**
have fun

### CODE REVIEW

File: node.rb [hunk 1]
```diff
@@ -1,21 +1,51 @@
...
+    @active = false
+    @distance_to_end = nil
+    @distance_to_start = nil
+  end
+
+  def has_value? value
+    v == value
+  end
+
+  def activate!
```
Comment (apellizzn) [inline node.rb]: Bang should be used only if the method could throw an exception / error

File: tree.rb [hunk 2]
```diff
@@ -1,10 +1,14 @@
 class Tree
-  attr_accessor :data
+  attr_accessor :data, :start, :finish
 
   def initialize(data)
     self.data = data
   end
 
+  def active
+    data.flatten.select{|n| n.active }
```
Comment (apellizzn) [inline tree.rb]: Can be rewritten like `data.flatten.select(&:active)`

**Comment by apellizzn** [comment]
`def set(node)` seems to be unused after your refactoring can be deleted

File: tree.rb [hunk 3]
```diff
@@ -1,37 +1,40 @@
...
-      get(node.r, node.c - 1),
-      get(node.r, node.c + 1),
-      get(node.r - 1, node.c),
-      get(node.r - 1, node.c + 1),
-      get(node.r - 1, node.c - 1),
-      get(node.r + 1, node.c),
-      get(node.r + 1, node.c + 1),
-      get(node.r + 1, node.c - 1),
-    ].compact.select(&:available?)
+    COORD.map {|coord| get(node.r + coord[0], node.c + coord[1]) }.compact.select(&:available?)
```
Comment (Steph186) [inline tree.rb:15]: also changed this!

**Review by apellizzn — APPROVED** [review]

**Pull request merged**